File: native_serivce/java_client/java/net/myocat/client/MainActivity.java
package net.myocat.client;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.os.IBinder;
import android.os.RemoteException;
import android.util.Log;

import net.myocat.ITestService;

import java.lang.reflect.Method;

public class MainActivity extends AppCompatActivity
{
    private static final String TAG = MainActivity.class.getSimpleName();

    private IBinder getService(String serviceName)
    {
        IBinder binder = null;
        try{
            Object object = new Object();
            Method getService = Class.forName("android.os.ServiceManager").getMethod("getService", String.class);
            Object obj = getService.invoke(object, new Object[]{new String(serviceName)});
            binder = (IBinder)obj;
        } catch(Exception e) {
            Log.d(TAG, e.toString());
        }

        return binder;
    }

    private ITestService mService;

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mService = ITestService.Stub.asInterface(getService("net.myocat.ITestService"));
        if (mService == null) {
            Log.d(TAG, "failed to getting service " + mService);
        }

        try {
            mService.echo("TTTTT");
        }
        catch (RemoteException e) {
            e.printStackTrace();
        }

    }
}

File: native_serivce/app/app/src/main/java/net/myocat/testapp/MainActivity.kt
package net.myocat.testapp

import android.content.ComponentName
import android.content.Context
import android.content.Intent
import android.content.ServiceConnection
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.IBinder

import net.myocat.ITestService;
import net.myocat.ITestServiceListener;

class MainActivity : AppCompatActivity() {

    companion object {
        private val TAG = MainActivity::class.java.simpleName
    }

    private var mTestService : ITestService ?= null

    private var mServiceConnection = object: ServiceConnection {
        override fun onServiceConnected(compName: ComponentName?, binder: IBinder?) {
            mTestService = ITestService.Stub.asInterface(binder)

            android.util.Log.d("PRAZEN", "==== Service " + mTestService);
            mTestService?.echo("1111");
        }

        override fun onServiceDisconnected(p0: ComponentName?) {
            mTestService = null
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var intent = Intent().also {
            it.setClassName("net.myocat.ITestService", "net.myocat.ITestService")}

        android.util.Log.d("PRAZEN", "Try bind Service");
        bindService(intent, mServiceConnection, Context.BIND_AUTO_CREATE)
    }
}